cid/helpers/quicksight/__init__.py

```diff
@@ -550,6 +559,16 @@ def delete_dashboard(self, dashboard_id):
         del self._dashboards[dashboard_id]
         return result
 
+    def delete_data_source(self, datasource_id):
+        """ Deletes an AWS QuickSight dashboard """
+        paramaters = {
+            'AwsAccountId': self.account_id,
+            'DataSourceId': datasource_id
+        }
+        logger.info(f'Deleting DataSource {datasource_id}')
+        result = self.client.delete_data_source(**paramaters)
+        return result
```

Comment:
We have to also remove it from the internal resources structure if it existed

Reply:
yes. fixed